flutter/lib/src/flutter_enricher_event_processor.dart

```diff
@@ -130,8 +129,18 @@ class FlutterEnricherEventProcessor extends EventProcessor {
     final tempDebugBrightnessOverride = debugBrightnessOverride;
     final initialLifecycleState = _window?.initialLifecycleState;
     final defaultRouteName = _window?.defaultRouteName;
+    final mode = kDebugMode
+        ? 'debug'
+        : kReleaseMode
+            ? 'release'
+            : 'profile';
```

Comment:
That's more useful if you set it as an `options#environment` during SDK init.

Reply:
Well environment can be something custom, this is the actual Flutter compile mode.
For example I can have `environment: staging` with a release mode or debug mode compile.

And yes you are right, it is being set in the Dart package into the `options.environment` but I can not find it anywhere in the Sentry UI.

Reply:
> And yes you are right, it is being set in the Dart package into the `options.environment` but I can not find it anywhere in the Sentry UI.

My own stupidity, I am overriding it with `staging`.

Reply:
Doesn't change the fact, that I need both bits in the event.
But this part could move into the pure `dart_context`.

Reply:
Since the `mode` is already set via the `environment` flag and you've changed yourself.
I don't see much benefit in adding it to the event by default, it'd be duplicated data for most of the users.

Also, I believe the majority of the events would come from `release` anyway, sounds like this bit of data you could add yourself using `Scope#setExtra` and friends.

@brustolin opinions?

Reply:
Take a lot at `_getRuntimes` as well, if the matter is release x debug, you can also see it from there, because of the Dart AOT or VM.

Reply:
The runtime is not available on Flutter for web, so that won't always work.

Reply:
I am using (and I assume others as well) environments across different projects for things like `alpha`, `beta`, `staging`, `production`.
I think that inferring an environment from the compile mode is an ok default. But the information should be visible somewhere else as well, especially since the default naming does not always imply a direct connection.

`kReleaseMode`, `--release` -> `production` -  I understand this now because I looked at the code but other users of the library may not dive this deep.